### PR TITLE
Additional documentation on binary deployments

### LINKED ISSUE #18: Update Binary Deployment Methods Documentation
Update the builds documentation to reference using a deployments folder in the application source which will copy binary artifacts to the `<JBOSS_HOME>/standalone/deployments`folder

### PR DESCRIPTION
**Pull request by sabre1041**
Documentation on placing binary content in deployments folder for S2I builds to resolve #18

### CODE REVIEW

**Comment by sabre1041** [comment]
Since I was already on the builds page, I also addressed #32

**Comment by sabre1041** [comment]
@etsauer Added a new section on the binary type of builds that was added in OSE 3.1

**Comment by etsauer** [comment]
@sabre1041 this looks good to me.

@oybed @trevorquinn @JaredBurck , second?

File: playbooks/app_dev/builds.adoc [hunk 1]
```diff
@@ -411,6 +417,79 @@ The following is a snippet of code demonstrating the use of a BuildConfig
...
+{
+ "source" : {
+    "type" : "Dockerfile",
+    "dockerfile": "FROM centos:7\nRUN yum install -y httpd"
+ },
+}
+
+----
+
+=== Binary
```
Comment (oybed) [inline playbooks/app_dev/builds.adoc]: I believe this is only applicable to newer versions of OpenShift? - if so, maybe add another "tag" to this section indicating the minimum version of OpenShift (if known?)

File: playbooks/app_dev/builds.adoc [hunk 2]
```diff
@@ -335,9 +337,13 @@ There are a number of existing S2I builders that you can look to leverage for yo
 
 === Binary deployments
 
-In certain cases, an application may be previously compiled outside of an OpenShift build. You can use the _assemble_ phase of an S2I build process to retrieve the previously packaged artifact and insert it into the typical build process of the builder being leveraged.
 
-The following is an example of a deployment of a binary application for the OpenShift JBoss EAP image
+In certain cases, an application may be previously compiled outside of an OpenShift build. There are two methods to include an existing binary artifact(s) in the resulting image:
+
+1. Customize the _assemble_ script as part of the S2I build process to retrieve a previously packaged artifact from a remote source (such as an artifact repository) and place in the JBoss deployments folder.
```
Comment (oybed) [inline playbooks/app_dev/builds.adoc]: Both of these methods seems specific to JBoss. Although that is most likely capturing the majority of our customers and engagements, could it be written a bit more generic? ... or potentially prefaced with a comment stating that this is JBoss specific, and a similar approach can also be used with other types of deployment?

Comment (etsauer) [inline playbooks/app_dev/builds.adoc]: The content here is generic to any language that can be compiled/built. What I would maybe recommend is more clearly separate the content into

- Generic guidance about how a binary deployment might be done (no refs to JBoss sepcifically)
- A working example using JBoss EAP

**Comment by trevorquinn** [comment]
I'm good with this and have a training need to have this updated as soon as possible. One change, however: in the assemble script example in the Binary Deployments section, the default assemble script is /usr/local/s2i/assemble, not /usr/local/sti/assemble.

**Comment by sabre1041** [comment]
@trevorquinn location of s2i scripts has been updated in the branch

**Pull request merged**